panels/map/ha-panel-map.html

```diff
@@ -3,6 +3,7 @@
 <link rel="import" href="../../bower_components/app-layout/app-toolbar/app-toolbar.html">
 
 <script src="../../bower_components/leaflet/dist/leaflet.js"></script>
+<script src="../../bower_components/leaflet-providers/leaflet-providers.js"></script>
```

Comment:
I am not a big fan of including this file as it adds 20kb to the size of what the phone has to load and parse. I think that instead we should allow people to configure url and options.

Reply:
I'm inclined to agree and initially implemented the feature exactly that way - the map configuration allowed you to specify a tile server URL and passed that to Leaflet instead of the CartoDB one.

Unfortunately that approach does not handle attribution well. I can manually detect various common tile providers based on the URL and automatically add appropriate attribution, but it isn't very robust.

After trying both approaches I think leaflet-providers gives a significant usability improvement:
- Attribution is automatic and correct for all providers
- There is a pre-made and attractive list of map providers that we can link to from the documentation
- The configuration is easier for the user (just enter a map name, rather than pasting a URL and replacing URL encoded characters manually)

From a fresh page load, leaflet-providers is only accounting for 20kb out of approx. 1800 kb total (0.01%) so I think it's worth using for the better experience it provides

Reply:
We're not going to include 20kb for a feature that 99% of the users will not use. Please rewrite it to allow passing in url and attribution via config.

Reply:
Since map is now optional - only user adding `map:` to the config will load it.

Reply:
Correct and the panel is 139KB before gzip. Adding 15% on top of that to make it configurable is not ok.

Reply:
So maybe load it on demand if user asked for it in the config?
Although for that it would have to be vulcanized and registered separately.

Or maybe load it from a hosted location? The user need internet access for nap tiles anyway.

Reply:
All assets that we depend on during runtime should be hosted. If we would host it, we would see the location of instances via referrers and we don't want that.

Reply:
Doesn't the tile provider see the referer anyway?

Reply:
Correct. But that doesn't mean that we should add more. Especially scripts would be dangerous as that means that compromising our servers means that one can have scripts added to every instance out there 🤔

Reply:
Ok, I will rewrite it so that the URL, attribution name and attribution URL are passed manually and no additional scripts are included